Objects/object.c

```diff
@@ -2129,3 +2129,75 @@ _Py_Dealloc(PyObject *op)
 #ifdef __cplusplus
 }
 #endif
+
+
+static struct {
+    PyObject ***array;
+    Py_ssize_t len;
+    Py_ssize_t alloc;
+} once_variables = {NULL, 0, 0};
+
+int
+_PyOnceVar_Set(PyObject **var, PyObject *value)
+{
+    assert(var != NULL);
+    assert(*var == NULL);
```

Comment:
Since calculating `value` is not an atomic operation (it can involve even importing) there is possible a race condition. If `*var` is not NULL, just decref `value` and return 0.